### PR TITLE
temurin@22 22.0.2 (new cask)

### PR DESCRIPTION
**Pull request by ntroncos**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [ ] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [ ] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/t/temurin@22.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    url "https://api.adoptium.net/v3/info/release_versions?release_type=ga&architecture=#{arch}&image_type=jdk&jvm_impl=hotspot&os=mac&page=0&page_size=1&project=jdk&sort_method=DEFAULT&sort_order=DESC&vendor=eclipse"
+    strategy :json do |json|
+      json["versions"].filter_map do |version|
+        match = version["openjdk_version"].match(/^(\d+(?:\.\d+)*)\+(\d+(?:\.\d+)*)(?:-LTS)?$/i)
+        next if match.blank?
+
+        "#{match[1]},#{match[2]}"
+      end
+    end
+  end
```
Comment (khipp) [inline Casks/t/temurin@22.rb:24]:
```suggestion
  livecheck do
    url "https://api.adoptium.net/v3/assets/feature_releases/#{version.major}/ga?architecture=#{arch}&image_type=jdk&jvm_impl=hotspot&os=mac&page=0&page_size=1&project=jdk&sort_method=DEFAULT&sort_order=DESC&vendor=eclipse"
    regex(/^jdk-(\d+(?:\.\d+)*?)\+(\d+(?:\.\d+)*)(?:-LTS)?$/i)
    strategy :json do |json, regex|
      json.map do |release|
        match = release["release_name"]&.match(regex)
        next if match.blank?

        "#{match[1]},#{match[2]}"
      end
    end
  end
```

**Review by khipp — COMMENTED** [review]
Thank you for your first contribution to `homebrew-cask`, @ntroncos! 🎉

We still need to update the `livecheck` to keep the major version fixed (see `temurin@21`).

**Comment by SMillerDev** [comment]
Is this one still supported upstream? It seems it is not: https://endoflife.date/eclipse-temurin

**Comment by ntroncos** [comment]
> Is this one still supported upstream? It seems it is not: https://endoflife.date/eclipse-temurin

Looks like we are in a bit of a bind. 
We are actively using Java 22, but are not quite ready to move to 23. 

Having it disappear really throws a wrench into all our dev environments.

**Comment by bevanjkay** [comment]
@ntroncos I would recommend you host this in your [own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) - as we shouldn't be adding versioned casks to `homebrew-cask` for software that is no longer supported upstream.

**Comment by ntroncos** [comment]
@bevanjkay checks out. 
I was not expecting the support for 22 to be dropped immediately by upstream, but it makes sense for them to not maintain it. 
Creating are own tap sounds like the path forward if we cannot aggressively move to java 23. 

Thanks for the guidance.

**Pull request closed without merging**